lib/cdo/redshift.rb

```diff
@@ -16,6 +41,54 @@ def initialize
   end
 
   def exec(sql_query)
-    @conn.exec(sql_query)
+    result = @conn.exec(sql_query)
+    # result_status is an undocumented method that returns the integer query status.
+    # res_status is an undocumented method that returns the string identifier for a given integer query status.
+    result_status = result.res_status(result.result_status)
+    raise PostgreSQLQueryError.new(result_status) unless QUERY_SUCCESS_CODES.include? result_status
+    result
+  end
+
+  # DMS Replication Tasks import data into temporary / staging tables prefixed with "_import_".
+  # Complete import of staged data from production database into Redshift by truncating each
+  # target table, moving all of the rows from each import table to its corresponding target table, and then dropping
+  # each temporary / staging import table.
+  def complete_table_import
+    Cdo::DMS::REDSHIFT_SCHEMAS_IMPORTED_FROM_DATABASE.each do |schema|
```

Comment:
Could we pass this set of constants as an argument to this method rather than hard-coding the list of tables from this constant?

Reply:
Done!